src/precice/impl/WriteDataContext.cpp

```diff
@@ -31,7 +31,7 @@ void WriteDataContext::resetData(bool atEndOfWindow, bool isTimeWindowComplete)
 
   if (isTimeWindowComplete && hasWriteMapping()) { // manually overwrite value at beginning with value from end. Need this exception for WriteDataContext with write mapping, because CouplingScheme is not able to update _providedData, if write mapping sits between _providedData and _toData. CouplingScheme in this case only has access to _toData.
     PRECICE_ASSERT(atEndOfWindow, "isTimeWindowComplete without atEndOfWindow is forbidden!");
-    auto atEnd = _providedData->timeStepsStorage().stamples().back().sample;
+    const auto &atEnd = _providedData->timeStepsStorage().stamples().back().sample;
```

Comment:
I did not read through the whole code, but considering that you are investigating performance, [this section about Eigen and `auto`](https://eigen.tuxfamily.org/dox/TopicPitfalls.html#title3) might be of interest.

Reply:
Thank you for the hint! That's very good to know, but looks to me that the problem is somewhere else. I changed the `auto` to `Eigen::VectorXd` at several places without a observable difference in runtime. However, we actually use `auto` at many places where we access `Data::values()` (e.g. `Acceleration::applyRelaxation`).